src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1909,13 +1935,44 @@ class DumperController : public CHeapObj<mtInternal> {
    const uint   _dumper_number;
    uint   _complete_number;
 
+   bool   _started; // VM dumper started and acquired global writer lock
+   Mutex* _global_writer_lock;
+
  public:
    DumperController(uint number) :
-     _lock(new (std::nothrow) PaddedMonitor(Mutex::safepoint, "DumperController_lock")),
+     _lock(new (std::nothrow) PaddedMonitor(Mutex::nosafepoint - 1, "DumperController_lock")),
      _dumper_number(number),
-     _complete_number(0) { }
+     _complete_number(0),
+     _started(false),
+     _global_writer_lock(new (std::nothrow) Mutex(Mutex::nosafepoint, "DumpWriter_lock"))
```

Comment:
We discussed a suggestion to place both lock definitions together:
 - lines: 1934 & 1939
 - lines: 1943 & 1947
 Then it will be a good place to add a comment at the start of `DumperController` constructor to explain why we use `MutexLocker` with the flag `Mutex::_no_safepoint_check_flag`.

Reply:
Done.